### DISCUSSION TITLE
call_next(request) isn't working when added inside an asyncio task.

### DISCUSSION THREAD

**Discussion by dhulmul** [fastapi/fastapi#2751] [Questions]
I have a setup with Nginx acting as proxy and a FastAPI upstream server. When a large file is uploaded the connection times out and Bad Gateway error is thrown. To fix this, I wrote a keep-alive middleware which will keep sending empty string as response until the request has been completed. Once the request is complete, I sent the actual response. Below is the code for the same (non-working). The part which doesn't work is `call_next(request)`. I get a multi-part decoder error `Did not find boundary character 130 at index 2` with the current code. If I put the `call_next(request)` in the __call__ method of the middleware then it works fine but then I won't be able to stream blank data to keep the connection alive.

Could someone point out why this issue happens and a possible fix?

```
class KeepAliveMiddleware: 

    @staticmethod
    async def slow_method():
        await asyncio.sleep(0.01)
        return "done slow"

    @staticmethod
    def send_blank(future):
        while not future.done():
            yield ""
        yield future.result()

    @staticmethod
    async def long_running_job(request, call_next):
        try:
            # await KeepAliveMiddleware.slow_method()  # This works fine
            return await call_next(request) # This leads to the error mentioned above.
        except Exception as ae:
            print(ae)

    async def __call__(self, request: Request, call_next):
        try:
            future = asyncio.create_task(
                KeepAliveMiddleware.long_running_job(request, call_next)
            )
            return StreamingResponse(KeepAliveMiddleware.send_blank(future))

        except Exception as e:
            print("Exception in task... ", e)


keep_alive_middleware = KeepAliveMiddleware()   
app.middleware("http")(keep_alive_middleware)
```

**Comment by tiangolo** [comment]
Maybe it would be better to write it in an issue that would guide you to follow the template, provide a complete, self-contained example code, etc. That way me or others can help you better.
